### PR TITLE
3 create event

### LINKED ISSUE #3: As a user I can add entries that have a specific start and end time

### PR DESCRIPTION
**Pull request by tjonganthony**
Fix #3 

Main changes:
- Changed Entry from interface to abstract class
- Changed Deadline to Event
- TaskListCard.fxml is reformatted. Ignore this.

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/Parser.java [hunk 1]
```diff
@@ -172,18 +194,18 @@ private static LocalDateTime getLocalDateTimeFromArgument(String dateTimeString,
...
        String title = unwrapOptionalStringOrEmpty(argsTokenizer.getPreamble());
        if (title.isEmpty()) {
            return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
        }
-
+ 
        try {
            return new AddCommand(title,
-                   getDeadlineFromArgument(argsTokenizer),
+        		   getStartTimeFromArgument(argsTokenizer),
```
Comment (joeleba) [inline src/main/java/seedu/address/logic/parser/Parser.java]: indentation

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 2]
```diff
@@ -17,68 +17,57 @@
...
-     * @throws IllegalValueException if any of the raw values are invalid
-     */
-    // TODO: Implement Add for other types of entry
-    /* public AddCommand(String title, LocalDateTime deadline, Set<String> tags, String description)
-            throws IllegalValueException {
-        final Set<Tag> tagSet = new HashSet<>();
-        for (String tagName : tags) {
-            tagSet.add(new Tag(tagName));
+        
+        //@@author A0126539Y
```
Comment (joeleba) [inline src/main/java/seedu/address/logic/commands/AddCommand.java]: should put this at the top of the method. I think the handbook did say 🤔

File: src/main/java/seedu/address/logic/parser/Parser.java [hunk 3]
```diff
@@ -128,25 +129,46 @@ private static String getTitleFromArgs(ArgumentTokenizer argsTokenizer) throws I
...
+    
+    /**
+     * Extracts the new entry's startTime from the add command's tag arguments
+     * string. Format: YYYY-MM-DD HH:MM
+     */
+    private static LocalDateTime getStartTimeFromArgument(ArgumentTokenizer argsTokenizer) throws IllegalValueException {
+        String startTime = unwrapOptionalStringOrEmpty(argsTokenizer.getValue(startPrefix));
+        
+        if (startTime.isEmpty()) {
+        	return null;
```
Comment (joeleba) [inline src/main/java/seedu/address/logic/parser/Parser.java]: indentation

File: src/main/java/seedu/address/model/PredicateBuilder.java [hunk 4]
```diff
@@ -118,24 +120,24 @@ public String toString() {
...
-                        return false;
-                    } else {
-                        return task.getDeadline().compareTo(startDate) >= 0;
-                    }
-
-                default:
-                    return false;
-            }
-        }
+        	if (entry instanceof Task) {
```
Comment (joeleba) [inline src/main/java/seedu/address/model/PredicateBuilder.java]: hmm i think you might be using tab instead of 4 spaces

**Review by joeleba — CHANGES_REQUESTED** [review]
Looks fine. Just need to change the indentations & use 4 spaces. Gonna have some conflicts when we merge in #67

File: src/main/java/seedu/address/logic/commands/AddCommand.java (hunk 2 above)
Comment (tjonganthony) [reply src/main/java/seedu/address/logic/commands/AddCommand.java]: Right, will fix.

**Comment by tjonganthony** [comment]
Fixed  problems from joe.

**Pull request merged**